src/Item.js

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2020 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+const _ = require('../custom-lodash');
+const isPlainObject = _.isPlainObject;
+const isEmpty = _.isEmpty;
+const omit = _.omit;
+
+const dataMatchesContraints = require('./utils/dataMatchesContraints');
+const ITEM_CONSTRAINTS = require('./ItemConstraints');
```

Comment:
I would keep the constraints within Item.js unless there is a size reduction benefit or if they are used elsewhere and sharing makes sense.